apps/openmw/mwsound/openal_output.cpp

```diff
@@ -20,39 +20,23 @@
 #include "soundmanagerimp.hpp"
 #include "loudness.hpp"
 
+#include "efx-presets.h"
+
 #ifndef ALC_ALL_DEVICES_SPECIFIER
 #define ALC_ALL_DEVICES_SPECIFIER 0x1013
 #endif
 
-#ifndef ALC_SOFT_HRTF
-#define ALC_SOFT_HRTF 1
-#define ALC_HRTF_SOFT                            0x1992
-#define ALC_DONT_CARE_SOFT                       0x0002
-#define ALC_HRTF_STATUS_SOFT                     0x1993
-#define ALC_HRTF_DISABLED_SOFT                   0x0000
-#define ALC_HRTF_ENABLED_SOFT                    0x0001
-#define ALC_HRTF_DENIED_SOFT                     0x0002
-#define ALC_HRTF_REQUIRED_SOFT                   0x0003
-#define ALC_HRTF_HEADPHONES_DETECTED_SOFT        0x0004
-#define ALC_HRTF_UNSUPPORTED_FORMAT_SOFT         0x0005
-#define ALC_NUM_HRTF_SPECIFIERS_SOFT             0x1994
-#define ALC_HRTF_SPECIFIER_SOFT                  0x1995
-#define ALC_HRTF_ID_SOFT                         0x1996
-typedef const ALCchar* (ALC_APIENTRY*LPALCGETSTRINGISOFT)(ALCdevice *device, ALCenum paramName, ALCsizei index);
-typedef ALCboolean (ALC_APIENTRY*LPALCRESETDEVICESOFT)(ALCdevice *device, const ALCint *attribs);
-#ifdef AL_ALEXT_PROTOTYPES
-ALC_API const ALCchar* ALC_APIENTRY alcGetStringiSOFT(ALCdevice *device, ALCenum paramName, ALCsizei index);
-ALC_API ALCboolean ALC_APIENTRY alcResetDeviceSOFT(ALCdevice *device, const ALCint *attribs);
-#endif
-#endif
-
 
 #define MAKE_PTRID(id) ((void*)(uintptr_t)id)
 #define GET_PTRID(ptr) ((ALuint)(uintptr_t)ptr)
 
 namespace
 {
 
+// The game uses 64 units per yard, or approximately 69.99125109 units per meter.
+// Should this be defined publically somewhere?
+constexpr float UnitsPerMeter = 69.99125109f;
```

Comment:
Looks like Visual Studio 2013 does not support constexpr.

Reply:
Do we want to keep support for MSVC 2013? Our win builds are the only official ones...

Reply:
According to [this page](https://blogs.msdn.microsoft.com/vcblog/2013/11/18/announcing-the-visual-c-compiler-november-2013-ctp/), ```constexpr``` should be (mostly) supported in the November 2013 CTP which can be used with VS2013. Is it possible to update AppVeyor? Might need to ask @ace13 and whoever else builds on Windows if that's a good idea.

Reply:
Ah, the 2013 CTP. I've used that for personal projects before, not the nicest of things to either install or use.
I'd rather see us drop 2013 support than tell people to install something that's literally named "Preview" to be able to compile.

Reply:
We should not be supporting Windows XP either.   In linux land, if the distro is EoL, then we don't support it. Same for Microsoft, if they EoL WinXP (and hopefully Vista), then we don't officially support it either.

Reply:
While I'm open to the idea of upgrading, is it worth it purely for 'constexpr'? Is there anything else we would gain? If we do want to go ahead, best take it to the forums to ask who's using 2013.

> We should not be supporting Windows XP either. In linux land, if the distro is EoL, then we don't support it. Same for Microsoft, if they EoL WinXP (and hopefully Vista), then we don't officially support it either.

Generally agreed, but there's two types of support - we can allow things to run, but that doesn't mean we need to be accountable if someone can't get it working or reports a bug. Something that isn't officially supported can still happen to work, for a while anyway.

Sure, there's not much usecase for XP but if you happened to have a really old computer that is only used offline, there's no harm in not upgrading it.

Reply:
Agreed, but if the machine is used offline then you're not likely to be running OpenMW now then are you? ;)

If 'constexpr' is the straw that broke the camel's back for WindowsXP support, then so be it. We shouldn't be held behind for that.

Reply:
This instance of using ```constexpr``` is far from necessary, so if it's that much of an issue I can just use plain ```const```. But it would still be a good idea to discuss it on the forums for when constant expressions come up elsewhere in the code base.

Reply:
In fact this exact issue was already discussed in [this PR](https://github.com/OpenMW/openmw/pull/1414) but no decision was made there. Personally I don't think XP support should be dropped just for `constexpr` but there's always the option of using `if (MSVC_VERSION LESS 1900) add_definitions(-Dconstexpr)` ;)